app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -450,17 +453,46 @@ class PatientSummaryScreenController @Inject constructor(
   }
 
   private fun showUpdatePhoneDialogIfRequired(events: Observable<UiEvent>): Observable<UiChange> {
-    val patientUuidStream = events
-        .ofType<PatientSummaryScreenCreated>()
+    val screenCreations = events.ofType<PatientSummaryScreenCreated>()
+
+    val showForInvalidPhone = screenCreations
         .map { it.patientUuid }
+        .switchMap { patientUuid ->
+          hasInvalidPhone(patientUuid)
+              .take(1)
+              .filter { invalid -> invalid }
+              .map { { ui: Ui -> ui.showUpdatePhoneDialog(patientUuid) } }
+        }
 
-    return Observables
-        .zip(patientUuidStream, configProvider.toObservable())
-        .filter { (_, config) -> config.isUpdatePhoneDialogEnabled }
-        .switchMap { (patientUuid, _) -> Observables.zip(phoneNumber(patientUuid), lastCancelledAppointment(patientUuid)) }
-        .filter { (number, appointment) -> appointment.updatedAt > number.updatedAt }
-        .take(1)
-        .map { (number) -> { ui: Ui -> ui.showUpdatePhoneDialog(number.patientUuid) } }
+    val showForMissingPhone = screenCreations
+        .filter { it.caller == SEARCH }
```

Comment:
@vinaysshenoy this controller depends on `PatientSummaryScreenKey#caller == SEARCH` to identify a subsequent visit by a patient. Do you think this will continue to work fine when the summary screen is opened by scanning a BP passport?

Reply:
Good point. It won't. Should we change it to exclude `NEW_PATIENT` instead?

Reply:
Yes. Sanchita is going to do it. Please coordinate?

https://github.com/simpledotorg/simple-android/pull/542#discussion_r265457166